bot/cogs/slowmode.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+from discord import TextChannel
+from discord.ext.commands import Cog, Context, group
+
+from bot.bot import Bot
+from bot.constants import Emojis, MODERATION_ROLES
+from bot.converters import DurationDelta
+from bot.decorators import with_role
+from bot.utils import time
+
+log = logging.getLogger(__name__)
+
+
+class Slowmode(Cog):
+    """Commands for getting and setting slowmode delays of text channels."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    @group(name='slowmode', aliases=['sm'], invoke_without_command=True)
+    async def slowmode_group(self, ctx: Context) -> None:
+        """Get and set the slowmode delay for a given text channel."""
+        await ctx.send_help(ctx.command)
+
+    @slowmode_group.command(name='get', aliases=['g'])
+    async def get_slowmode(self, ctx: Context, channel: TextChannel) -> None:
+        """Get the slowmode delay for a given text channel."""
+        delay = relativedelta(seconds=channel.slowmode_delay)
+
+        try:
+            humanized_delay = time.humanize_delta(delay, precision=3)
+
+        except TypeError:
+            # The slowmode delay is 0 seconds,
+            # which causes `time.humanize_delta` to raise a TypeError
+            humanized_delay = '0 seconds'
```

Comment:
Looking over the `humanize_delta` source, it wants a string for `precision`.

It will be matched against these:

```python
units = (
    ("years", delta.years),
    ("months", delta.months),
    ("days", delta.days),
    ("hours", delta.hours),
    ("minutes", delta.minutes),
    ("seconds", delta.seconds),
)
```

The `TypeError` comes from here and is a result of an incorrect type being passed in:

https://github.com/python-discord/bot/blob/74de316657d5fed9d5474fa23d05cbe09d9ec470/bot/utils/time.py#L25-L26

Reply:
I don't know how I missed this. I'll fix that up 👍